### ISSUE TITLE
Why use keras predict_proba to predict the probability has the same value?

### ISSUE THREAD

**Issue by dapsjj** [keras-team/keras#14259]
I use keras2.3.1.
The data in my csv likes this:
```
[
[0, 0,……, 0],
[0, 1,……, 0],
[0, 2,……, 0],
[0, 3,……, 0],
[1, 0,……, 0],
[2, 0,……, 0],
[3, 0,……, 0],
[1, 1,……, 1],
[2, 1,……, 1],
[1, 2,……, 1],
[3, 1,……, 1],
]
```

The training set has 400,000 rows data and the test set has 200,000 rows data.

The value of y_train is only 0 and 1, and there are more than 40 columns of data in csv.

I use this code to predict the probability of 1 with x_test,but all the probability in the result are the same value, why?

```
import pandas as pd
from keras.models import Sequential
from keras.layers import Dense
from sklearn.preprocessing import LabelEncoder
from keras.utils import np_utils

train = pd.read_csv(r'train.csv')
test = pd.read_csv(r'test.csv')

x_train = train[:, :-1]
y_train = train[:, -1]

encoder = LabelEncoder()
encoder.fit(y_train)
encoded_y = encoder.transform(y_train)
y_train = np_utils.to_categorical(encoded_y)
x_test = test

model = Sequential()
model.add(Dense(512, activation='relu', input_dim=48))
model.add(Dense(256, activation='relu'))
model.add(Dense(128, activation='relu'))
model.add(Dense(64, activation='relu'))
model.add(Dense(32, activation='relu'))
model.add(Dense(2, activation='softmax'))
model.compile(optimizer='rmsprop',
              loss='binary_crossentropy',
              metrics=['binary_accuracy'])

model.fit(x_train, y_train, epochs=15, batch_size=50000, verbose=1)
predict = model.predict_proba(x_test, batch_size=50000)
print(predict[:,1]) #same value
```

**Comment by Sanjaykrishnamurthy** [comment]
Yes, I too have the same issue.

For me, Keras Sequential works perfectly but the functional style predicts the same class, even for the training data.

The Functional style code below:
`
il = Input(shape=(7,))
d1 = Dense(128, activation='relu')(il1)
d2 = Dense(64, activation='relu')(d1)
d3 = Dense(32, activation='relu')(d2)
output = Dense(1, activation='softmax')(il)

model = Model(inputs=il, outputs=output)

opt = optimizer=keras.optimizers.Adam(1e-3)
loss = tf.keras.losses.BinaryCrossentropy(from_logits=True)


model.compile(optimizer=opt,
              loss=loss,
              metrics=['accuracy'])


X_train1 = X_train1.astype('float32')
y_train = y_train.astype('float32')

model.fit(X_train1, y_train,
          epochs=10,
          validation_split=.10,
          shuffle=True,
          batch_size=64)

y_pred = model.predict(X_train1)
roc_auc_score(y_train, y_pred)`

0.5


`np.unique(y_pred.flatten())`

array([1.], dtype=float32)

The sequential syle of the same logic above,
`model = keras.Sequential(
    [
        Dense(128, activation='relu'),
        Dense(64, activation='relu'),
        Dense(32, activation='relu'),
        Dense(1, activation="sigmoid")
    ]
)

opt = optimizer=keras.optimizers.Adam(1e-2)
loss = tf.keras.losses.BinaryCrossentropy(from_logits=True)


model.compile(optimizer=opt,
              loss=loss,
              metrics=['accuracy'])


X_train1 = X_train1.astype('float32')
y_train = y_train.astype('float32')

model.fit(X_train1, y_train,
          epochs=1,
          validation_split=.10,
          shuffle=True,
          batch_size=1024)

y_pred = model.predict(X_train1)
roc_auc_score(y_train, y_pred)`
0.42617482810730

`np.unique(y_pred.flatten())`
array([0.0000000e+00, 1.1756790e-38, 1.1756970e-38, ..., 1.9908031e-14,
       2.0582059e-14, 2.4092256e-14], dtype=float32)


I tried a simpler model with no hidden layers and only a sigmoid, at last, even then functional style predicts the same class.
`tf.__version__` 
'2.3.1'

I suspect it might be an issue with Keras functional style.

**Comment by Saduf2019** [comment]
@dapsjj 
https://github.com/keras-team/keras/issues/14233
